compaction/compaction_strategy.cc

```diff
@@ -86,7 +86,9 @@ std::optional<sstring> compaction_strategy_impl::get_value(const std::map<sstrin
     return it->second;
 }
 
-compaction_strategy_impl::compaction_strategy_impl(const std::map<sstring, sstring>& options) {
+compaction_strategy_impl::compaction_strategy_impl(compaction_manager& cm, const std::map<sstring, sstring>& options)
+    : _compaction_manager(&cm)
```

Comment:
This is entangling things too much together. The compaction strategy is used by compaction_manager to decide what to compact, and now the dependency is circular. It's making everything more difficult.

Reply:
Yes, but the strategy is a leaf (at least relatively) on the dependency graph now. compaction_manager is near the root. Let's not make it circular with such large span.